keyring.go

```diff
@@ -163,6 +163,15 @@ func (k KeyRing) VerifyJSONs(ctx context.Context, requests []VerifyJSONRequest)
 	if err != nil {
 		return nil, err
 	}
+
+	// TODO: we should distinguish here between expired keys, and those we don't have.
+	// If the key has expired, it's no use re-requesting it.
```

Comment:
Random Q: if the key is supplied by another server (eg perspective or key query API) does the original server sign when the key expires? Else you could get a form of censorship if the intermediary server changes the expiration time maliciously.

I know the federation API signs point to point requests, but won't that be stripped away by the intermediary server? E.g A <---> B <---> C is all signed, but A <-----> C is not